src/rpc/mining.cpp

```diff
@@ -249,8 +249,6 @@ UniValue prioritisetransaction(const JSONRPCRequest& request)
             + HelpExampleRpc("prioritisetransaction", "\"txid\", 0.0, 10000")
         );
 
-    LOCK(cs_main);
```

Comment:
Why is this being removed?  It's not obvious to me that this wouldn't result in some kind of race with AcceptToMemoryPool.

Generally I think of cs_main as required to make any changes to the mempool, while mempool.cs is sufficient for read-only access to the mempool...  `PrioritiseTransaction` changes the mempool (re-sorting it), though I guess doesn't add or remove anything -- so it's not clear to me whether this would be problematic or not, with our current code.  But it seems safer to me to not remove the lock here, because even if there's no race condition now, it seems like it could result in a future bug.

Reply:
Yes I can remove that commit, it's not entirely related to this PR. I guess it can be revisited later after annotations are in place.

Reply:
I'd prefer if we document such assumptions in the code. Preferably with a clang static annotation but maybe also with a short description about the locking requirements for that method/class, if needed.